### QUESTION 1

What are the main influences on Property Prices in Melbourne?

In this question, we are investigating the key attributes that contribute to deciding property prices. It considers dependent variables such as land size, building size, number and types of rooms, parking availability, property type, year built and time of sale. This will provide valuable insight to buyers, understanding what trade-offs may affect their budget as they look to enter the market, alongside providing sellers with some prediction modelling, basing their expected sale prices upon the attributes relevant to their property. 

In [2]:
#Importing the required packages
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as st
import numpy as np

In [3]:
#Save the dataset file path
housing_path = "Property Sales of Melbourne City.csv"

In [4]:
#Read the csv file into a dataframe
housing_df = pd.read_csv(housing_path)

In [5]:
#Display the data table for preview
housing_df.head()

Unnamed: 0      Suburb           Address  Rooms Type    Price Method  \
0           1  Abbotsford      85 Turner St      2    h  1480000      S   
1           2  Abbotsford   25 Bloomburg St      2    h  1035000      S   
2           4  Abbotsford      5 Charles St      3    h  1465000     SP   
3           5  Abbotsford  40 Federation La      3    h   850000     PI   
4           6  Abbotsford       55a Park St      4    h  1600000     VB   

  SellerG       Date  Distance  ...  Bathroom  Car  Landsize  BuildingArea  \
0  Biggin  3/12/2016       2.5  ...       1.0  1.0     202.0           NaN   
1  Biggin  4/02/2016       2.5  ...       1.0  0.0     156.0          79.0   
2  Biggin  4/03/2017       2.5  ...       2.0  0.0     134.0         150.0   
3  Biggin  4/03/2017       2.5  ...       2.0  1.0      94.0           NaN   
4  Nelson  4/06/2016       2.5  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea  Lattitude Longtitude             Regionname  \
0        NaN        Yarra   -37.7996   144.9984  Northern Metropolitan   
1     1900.0        Yarra   -37.8079   144.9934  Northern Metropolitan   
2     1900.0        Yarra   -37.8093   144.9944  Northern Metropolitan   
3        NaN        Yarra   -37.7969   144.9969  Northern Metropolitan   
4     2014.0        Yarra   -37.8072   144.9941  Northern Metropolitan   

   Propertycount  
0         4019.0  
1         4019.0  
2         4019.0  
3         4019.0  
4         4019.0  

[5 rows x 22 columns]

In [7]:
#Provide a concise summary of the dataframe
housing_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18396 entries, 0 to 18395
Data columns (total 22 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     18396 non-null  int64  
 1   Suburb         18396 non-null  object 
 2   Address        18396 non-null  object 
 3   Rooms          18396 non-null  int64  
 4   Type           18396 non-null  object 
 5   Price          18396 non-null  int64  
 6   Method         18396 non-null  object 
 7   SellerG        18396 non-null  object 
 8   Date           18396 non-null  object 
 9   Distance       18395 non-null  float64
 10  Postcode       18395 non-null  float64
 11  Bedroom2       14927 non-null  float64
 12  Bathroom       14925 non-null  float64
 13  Car            14820 non-null  float64
 14  Landsize       13603 non-null  float64
 15  BuildingArea   7762 non-null   float64
 16  YearBuilt      8958 non-null   float64
 17  CouncilArea    12233 non-null  object 
 18  Lattit

In [8]:
#Remove the old indexing column 'Unnamed: 0'
del housing_df["Unnamed: 0"]

#Check the changes have applied
housing_df.head()

Suburb           Address  Rooms Type    Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [9]:
#Summary Table of the numeric variables
housing_df.describe()

Rooms         Price      Distance      Postcode      Bedroom2  \
count  18396.000000  1.839600e+04  18395.000000  18395.000000  14927.000000   
mean       2.935040  1.056697e+06     10.389986   3107.140147      2.913043   
std        0.958202  6.419217e+05      6.009050     95.000995      0.964641   
min        1.000000  8.500000e+04      0.000000   3000.000000      0.000000   
25%        2.000000  6.330000e+05      6.300000   3046.000000      2.000000   
50%        3.000000  8.800000e+05      9.700000   3085.000000      3.000000   
75%        3.000000  1.302000e+06     13.300000   3149.000000      3.000000   
max       12.000000  9.000000e+06     48.100000   3978.000000     20.000000   

           Bathroom           Car       Landsize  BuildingArea    YearBuilt  \
count  14925.000000  14820.000000   13603.000000   7762.000000  8958.000000   
mean       1.538492      1.615520     558.116371    151.220219  1965.879996   
std        0.689311      0.955916    3987.326586    519.188596    37.013261   
min        0.000000      0.000000       0.000000      0.000000  1196.000000   
25%        1.000000      1.000000     176.500000     93.000000  1950.000000   
50%        1.000000      2.000000     440.000000    126.000000  1970.000000   
75%        2.000000      2.000000     651.000000    174.000000  2000.000000   
max        8.000000     10.000000  433014.000000  44515.000000  2018.000000   

          Lattitude    Longtitude  Propertycount  
count  15064.000000  15064.000000   18395.000000  
mean     -37.809849    144.996338    7517.975265  
std        0.081152      0.106375    4488.416599  
min      -38.182550    144.431810     249.000000  
25%      -37.858100    144.931193    4294.000000  
50%      -37.803625    145.000920    6567.000000  
75%      -37.756270    145.060000   10331.000000  
max      -37.408530    145.526350   21650.000000

In [13]:
#List of object type variables
object_columns = housing_df.select_dtypes("object")

#Summary Array of the object variables
object_columns.nunique()

Suburb           330
Address        18134
Type               3
Method             5
SellerG          305
Date              58
CouncilArea       33
Regionname         8
dtype: int64

This dataset contains 18396 entries, with each entry pertaiing to 1 property sold in Melbourne. There are 21 variables included in this dataset:

1. 'Suburb' - The suburb the 